cmd/agent/app/flare.go

```diff
@@ -86,7 +86,7 @@ func makeFlare(caseID string) error {
 	}
 
 	pprofURL = fmt.Sprintf("http://127.0.0.1:%s/debug/pprof", config.Datadog.GetString("expvar_port"))
```

Comment:
What if we change `CreatePerformanceProfile` to accept (profileURL, profilingDuration, cpuDuration), and invoke it twice, once as it already is for core agent with the `expvar_port` in the url and the cpuDuration=profilingDuration, and then if `apm_config.enabled` we invoke it with `apm_config.receiver_port` and `4` as cpuDuration. That way, we do not need specialized logic for core and trace agent in `CreatePerformanceProfile`. 

I guess we would also have to create and pass in the profile struct before, so we only have to create it once and populate it with the necessary profile data. Does this make sense to you?

What if we check here whether `apm_config.enabled`,

Reply:
Yep, I like that. I think I made the change IIUC.